config/value.go

```diff
@@ -337,11 +338,17 @@ func convertValue(value interface{}, targetType reflect.Type) (interface{}, erro
 	}
 	switch v := value.(type) {
 	case string:
-		switch targetType.Name() {
-		case "int":
+		target := reflect.New(targetType).Interface()
+		switch target.(type) {
```

Comment:
do a colon equals assignment here like line 338 so you don't have to re-cast on 348